systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.utils.kafkaUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.templates.crd.KafkaTopicTemplates;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import io.strimzi.systemtest.resources.ResourceManager;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import io.strimzi.systemtest.enums.CustomResourceStatus;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static io.strimzi.systemtest.Constants.INFRA_NAMESPACE;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+
+@SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+public class KafkaTopicScalabilityUtils {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaTopicUtils.class);
+    protected static ResourceManager resourceManager = ResourceManager.getInstance();
+
+    public static void createTopicsViaK8s(ExtensionContext extensionContext, String clusterName, String topicPrefix,
+                                          int numberOfTopics, int numberOfPartitions, int numberOfReplicas, int minInSyncReplicas) {
+        LOGGER.info("Creating topics via Kubernetes");
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopicName = topicPrefix + i;
+            LOGGER.debug("Creating topic {}", currentTopicName);
+
+            CompletableFuture.runAsync(() ->
+                    resourceManager.createResource(extensionContext, false, KafkaTopicTemplates.topic(
+                            clusterName, currentTopicName, numberOfPartitions, numberOfReplicas, minInSyncReplicas, INFRA_NAMESPACE).build()));
+        }
+    }
+
+    public static void checkTopicsState(String topicPrefix, int numberOfTopics, Enum<?> state) {
```

Comment:
You are not "checking" the status, you are "waiting" for the status. So it should have proper name.